### ISSUE TITLE
multi-modality (CT-MRI) image registration for Head and neck，

### ISSUE THREAD

**Issue by shchojj** [voxelmorph/voxelmorph#201]
Hello Adrian,
I’m using VXM for multi-modality (CT-MRI) image registration. In the beginning, I use the loss is [MI](https://github.com/voxelmorph/voxelmorph/blob/ed94aa2c7f43565ef5220a087c1fa4b072348b51/voxelmorph/tf/losses.py#L318). I am not sure whether the Bins(bins = np.arange(0, 32, 1, dtype=np.float32)) is set correctly and the data is normalized. But MI is super heavy on GPU memory, so I gave it up. 
Then see the [MIND](https://github.com/voxelmorph/voxelmorph/pull/145#issue-372397584), it friendly to GPU memory, and seems to be effective.
However, my data are real clinical data, that is, MRI images are incomplete and tilt at any angle after rigid registration . 
![Snipaste_2020-06-22_11-12-20](https://user-images.githubusercontent.com/7993631/85245289-b8a3b380-b479-11ea-90f3-3663147a8932.png)

After registration with MIND ( tf version with diffeomorphic), I found that after deformation, MRI images would be forced to match to CT images, perhaps these pixels are not in MRI at all. I can't guarantee that the rigid body transformation is completely correct, so I can't just calculate where the MRI corresponds to the CT. Also because some of the data was too small I used the same size but difference spacing to ensure that the data I was cropping out would be padding the volume of fixed size.
![Snipaste_2020-06-22_11-19-17](https://user-images.githubusercontent.com/7993631/85245476-4089bd80-b47a-11ea-85e8-0d9235e21d89.png)

Do you have any suggestions to make my deformation small and correct, provided that the result of rigid body registration is robust?

**Comment by adalca** [comment]
Hi @shchojj , really sorry for missing this in the summer. How are you doing with this, did you make progress?
are you still stuck?

BTW, you said you used MIND with tf but I don't think we had that released, did we? Did you code MIND for tf?

**Comment by shchojj** [comment]
@adalca Thank you for your reply. With reference to previous pytorch code (https://github.com/voxelmorph/voxelmorph/pull/145/commits/389274436263ef0cd9a32a566f12bf7f048f6843), and modified to tf version. MIND has been successfully used in CT-MR. Because it is head and neck data, the deformation is relatively small, but MR data is incomplete (the slice number is very small), so it relies heavily on rigid registration.
However, the known loss function is not applicable to multi-mode rigid registration, especially for large rigid displacement. Therefore, I returned to the traditional optimization algorithm and now the registration has been completed.
Thanks to Voxelmorph for helping me solve the actual clinical problems, I hope it will become more perfect ~
